### PR TITLE
Fix empty join token ui state

### PR DESCRIPTION
**Pull request by avatus**
Empty join token responses would show an error instead of an empty list message

Screenshot 2024-08-13 at 8 41 35 AM

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Comment by avatus** [comment]
Backport collected here https://github.com/gravitational/teleport/pull/44701

File: web/packages/teleport/src/services/joinToken/joinToken.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ class JoinTokenService {
   fetchJoinTokens(signal: AbortSignal = null): Promise<{ items: JoinToken[] }> {
     return api.get(cfg.getJoinTokensUrl(), signal).then(resp => {
       return {
-        items: resp.items.map(makeJoinToken),
+        items: resp.items?.map(makeJoinToken) || [],
```
Comment (ravicious) [inline web/packages/teleport/src/services/joinToken/joinToken.ts:76]: Does the API endpoint not include `items` when there are no items to return?

Comment (marcoandredinis) [reply web/packages/teleport/src/services/joinToken/joinToken.ts:76]: It includes it, but it's `null`:
image

Comment (avatus) [reply web/packages/teleport/src/services/joinToken/joinToken.ts:76]: Looks like it has to do with [not using zero-length slices](https://stackoverflow.com/questions/56200925/return-an-empty-array-instead-of-null-with-golang-for-json-return-with-gin).

**Pull request merged**